apps/api/migrations/integration-scheme-update/add-primary-priority-migration.ts

```diff
@@ -0,0 +1,139 @@
+      const integrations = await integrationRepository.find(
+        {
+          _organizationId: organization._id,
+          _environmentId: environment._id,
+          channel,
+          active: true,
+        },
+        undefined,
+        { sort: { createdAt: -1 } }
+      );
+      console.log('------');
+      console.log(`Found ${integrations.length} active integrations for the ${channel} channel`);
+
+      for (let i = 0; i < integrations.length; i++) {
+        const integration = integrations[i];
+        const payload = {
+          primary: i === 0,
+          priority: integrations.length - i,
+        };
+        console.log(`${i + 1}. Updating integration "${integration.name}" with: `, payload);
+        await integrationRepository.update(
+          {
+            _id: integration._id,
+            _environmentId: integration._environmentId,
+            _organizationId: integration._organizationId,
+          },
+          {
+            $set: payload,
+          }
+        );
+      }
+
+      const inactiveIntegrations = await integrationRepository.find({
+        _organizationId: organization._id,
+        _environmentId: environment._id,
+        channel,
+        active: false,
+      });
+      console.log('------');
+      console.log(`Found ${inactiveIntegrations.length} inactive integrations for the ${channel} channel`);
+
+      for (let i = 0; i < inactiveIntegrations.length; i++) {
+        const integration = inactiveIntegrations[i];
+        const payload = {
+          primary: false,
+          priority: 0,
+        };
+        console.log(`${i + 1}. Updating inactive integration "${integration.name}" with: `, payload);
+        await integrationRepository.update(
+          {
+            _id: integration._id,
+            _environmentId: integration._environmentId,
+            _organizationId: integration._organizationId,
+          },
+          {
+            $set: payload,
+          }
+        );
+      }
+    }
+
+    for (const channel of [ChannelTypeEnum.IN_APP, ChannelTypeEnum.PUSH, ChannelTypeEnum.CHAT]) {
+      const integrations = await integrationRepository.find({
+        _organizationId: organization._id,
+        _environmentId: environment._id,
+        channel,
+      });
+      console.log('------');
+      console.log(`Found ${integrations.length} integrations for the ${channel} channel`);
+
+      for (let i = 0; i < integrations.length; i++) {
+        const integration = integrations[i];
+        const payload = {
+          primary: false,
+          priority: 0,
+        };
+        console.log(`${i + 1}. Updating integration "${integration.name}" with: `, payload);
+        await integrationRepository.update(
+          {
+            _id: integration._id,
+            _environmentId: integration._environmentId,
+            _organizationId: integration._organizationId,
+          },
+          {
+            $set: payload,
+          }
+        );
+      }
```

Comment:
We don't support primary and priority for the In-App, Push, Chat. Because one subscriber can have multiple channels assigned, ex. two Push channels for two devices that are both "active" - so "primary" and with the highest "priority".